components/Auth/LoginRegDisplay.tsx

```diff
@@ -0,0 +1,56 @@
+"use client"
+
+import Link  from "next/link";
+import { signIn, signOut, useSession } from "next-auth/react"
+import { useEffect, useState } from "react";
+import { Button } from "../ui/button";
+
+const LoginRegDisplay = ()=>{
+    const {status} = useSession();
+    const [show,setShow] = useState(false);
+    const [loginBtnShow,setLoginBtnShow] = useState(false)
+    
+    useEffect(()=>{
```

Comment:
I would like to propose an enhancement to our current implementation of hook blocks, which are currently used extensively across multiple pages. One potential improvement would be to centralize this functionality by creating a dedicated page with a hook controller. This approach would allow us to pass the specific page to render as a prop, thereby eliminating redundant code blocks throughout our application.

To facilitate this improvement, I suggest creating an improvisation ticket to further explore and implement this enhancement. I am willing to investigate and contribute to this initiative if necessary.

Please let me know your thoughts on this proposal @VijeshVS

Reply:
Good idea!! I will look into this.
Just to be clear you meant to move the session logic into a separate file and define it as a hook to avoid repetition.

Reply:
Yes, even that logic works ig